src/variables.rs

```diff
@@ -0,0 +1,72 @@
+use yultsur::yul::{Identifier, IdentifierID};
+
+use crate::smt::{self, SMTExpr, SMTSort, SMTStatement, SMTVariable};
+use crate::ssa_tracker::SSATracker;
+
+pub struct Variable {
+    pub identifier: Identifier,
+    pub domain: Vec<SMTSort>,
+    pub codomain: SMTSort,
+    pub default_value: Option<SMTExpr>,
+}
+
+impl Variable {
+    pub fn smt_var(&self, ssa: &mut SSATracker) -> SMTVariable {
+        ssa.to_smt_variable(&self.identifier)
+    }
+
+    #[must_use]
+    pub fn generate_definition(&self, ssa: &mut SSATracker) -> SMTStatement {
+        let smt_var = ssa.introduce_variable_of_type(
+            &as_declaration(self.identifier.clone()),
+            self.codomain.clone(),
+        );
+        if let Some(default) = &self.default_value {
+            smt::define_fun(smt_var, self.domain.clone(), default.clone())
+        } else {
+            smt::declare_fun(smt_var, self.domain.clone())
+        }
+    }
+}
+
+/// Declares a new SSA index for the given variable and sets it to
+/// "unknown" if the condition is false and to the old value if it is true.
+#[must_use]
+pub fn havoc_unless(
```

Comment:
Do you think these should better be member functions?

Reply:
Might be cleaner when reading yea